[Classic] Add improvements for module scripting and other

hawkeye116477 opened the pull request:
* Improvements and fixes for module type scripting
* Removed some not needed deprecation warnings
* Restored possibility to install extensions for FF 57
* Performance improvement for atob()
* Fixes for Streams API (not yet ready to be enabled by default)
* Fixes for promises
* Implemented RegExp .matchAll (ES 2020) - https://www.reddit.com/r/waterfox/comments/ip5f85/cant_access_a_secured_gdrive_website/
* Implemented RegExp .dotAll (ES 2018) - https://github.com/MrAlex94/Waterfox/issues/1738#issuecomment-686955770
* Exposed Intl.PluralRules - #1749
* Reverted and applied again some changes which caused debug build bustage (debug part of Bug 1618131 is incomplete for now, Bug 1448589 needs to be ported for that)
* Disabled getRootNode() (function related to WebComponents to prevent mis-detection).
* Implemented ResizeObserver
* Implemented percentage for CSS opacity keywords - #1734
* Undraggable images shouldn't be selected by single clicking now (fixes problem mentioned on https://www.reddit.com/r/waterfox/comments/hfapnk/issue_on_stendecio_website/)
* Implemented `overflow-wrap: anywhere`.

aeiouaeiouaeiouaeiouaeiouaeiou commented:
I'm curious why this still hasn't been merged.

MrAlex94 commented:
A lot of changes that need testing, didn't have time to push for this update cycle.

hawkeye116477 commented:
Small update: 
I added fixes for regressions caused by percentage opacity and Resize Observer implementation made by Moonchild Team.
I also added fix for problem mentioned on https://www.reddit.com/r/waterfox/comments/hfapnk/issue_on_stendecio_website/.
